### PR TITLE
keep zipfile open during export

### PR DESCRIPTION
**Pull request by snopoke**
There is an exponential slow down for re-opening the zipfile and appending to it for each new blob. Running the export for Applications took ~24 hours for 256 docs using the append mode and came to ~3GB. Keeping the zipfie open cut the dump time down to 10mins and the file size down to 2.6GB.

@emord @millerdev 
@gcapalbo

### CODE REVIEW

**Comment by czue** [comment]
how bout that. the more you know.

**Review by czue — APPROVED** [review]

File: corehq/blobs/migrate.py [hunk 1]
```diff
@@ -462,12 +465,12 @@ def migrate(self, filename=None, reset=False, max_retry=2, chunk_size=100):
 
         migrator = self.migrator_class(self.slug, self.domain)
 
-        with allow_form_processing_queries():
-            for model_class, queryset in get_all_model_querysets_for_domain(self.model_class, self.domain):
-                for obj in queryset.iterator():
-                    migrator.process_object(obj)
+        with migrator:
+            with allow_form_processing_queries():
```
Comment (millerdev) [inline corehq/blobs/migrate.py]: no need to change, but an indentation level can be eliminated here:
```py
with migrator, allow_form_processing_queries():
```

File: corehq/blobs/zipdb.py [hunk 2]
```diff
@@ -38,8 +38,17 @@ def get_path(self, identifier=None, bucket=DEFAULT_BUCKET):
...
+
+    @property
+    def zipfile(self):
+        if self._zipfile is None:
+            self._zipfile = zipfile.ZipFile(self.zipname, 'w', allowZip64=True)
+        return self._zipfile
+
+    def close(self):
+        if self._zipfile:
+            self._zipfile.close()
```
Comment (millerdev) [inline corehq/blobs/zipdb.py]: Ha, I recommended something similar a while ago https://github.com/dimagi/commcare-hq/pull/13478/files/8902c1acdbc07d4d8565b830485e3a3e52c21997#r82321461

**Review by millerdev — APPROVED** [review]

File: corehq/blobs/migrate.py (hunk 1 above)
Comment (snopoke) [inline corehq/blobs/migrate.py]: Oh yea, always forget about that.

File: corehq/blobs/zipdb.py (hunk 2 above)
Comment (snopoke) [inline corehq/blobs/zipdb.py]: Yes, that was one of the reasons I even thought to look here.

Comment (emord) [inline corehq/blobs/zipdb.py]: Hmm. yeah. I didn't realize that it would take that long 24 h to 10 min seems unreal.

**Comment by benrudolph** [comment]
wow 😲

**Pull request merged**